bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/model/PartitionState.java

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.model;
+
+/**
+ * The {@link Partition} Paradox partition states. Retrieved and parsed from RAM memory responses.
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public class PartitionState {
+
+    private boolean isArmed;
+    private boolean isArmedInAway;
+    private boolean isArmedInStay;
+    private boolean isArmedInNoEntry;
+    private boolean isInAlarm;
+    private boolean isInSilentAlarm;
+    private boolean isInAudibleAlarm;
+    private boolean isInFireAlarm;
+
+    private boolean isReadyToArm;
+    private boolean isInExitDelay;
+    private boolean isInEntryDelay;
+    private boolean isInTrouble;
+    private boolean hasAlarmInMemory;
+    private boolean isInZoneBypass;
+
+    private boolean hasZoneInTamperTrouble;
+    private boolean hasZoneInLowBatteryTrouble;
+    private boolean hasZoneInFireLoopTrouble;
+    private boolean hasZoneInSupervisionTrouble;
+
+    private boolean isStayInstantReady;
+    private boolean isForceReady;
+    private boolean isBypassReady;
+    private boolean isInhibitReady;
+    private boolean areAllZoneclosed;
+
+    public String getMainState() {
+        return isArmed | isArmedInAway | isArmedInNoEntry | isArmedInStay ? "Armed" : "Disarmed";
+    }
+
+    public String getAdditionalState() {
+        StringBuilder sb = new StringBuilder();
+        // TODO check if isInAlarm also includes the other three if yes -> check only the other three

```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
Removed. It's definitely not something important right now...